packages/realm_generator/test/common.realm.dart

```diff
@@ -0,0 +1,2 @@
+// MOCK FILE!
```

Comment:
I think this needs more detailed code comment. I know there's a comment in the PR, but in 2 years, it will take quite a bit of detective work to uncover the reason behind these mocks.

Reply:
Actually, your IDE will light up like a Christmas tree, if you try to remove one. 

In a way it was odd that the shared-part builder we used before did not require the part declaration. Now that it is required, the LSP will flag the part directive if the file is missing. 

So in a sense, I would need to explain why they weren't needed before, which seems less useful to the future developer (/me).

Reply:
Right, my point is that for someone who's not especially proficient with generators, it may not be obvious how the tests are setup and what role these mock files play. We don't have be overly descriptive, but a short comment like "These are here to avoid the IDE reporting errors during development. The actual file will be generated during the test run and compared to the .expected file" would to a long way to clarifying things to people new to the codebase.

Reply:
Would you prefer to repeat the description in each mock file, or add it as a common comment somewhere?

Reply:
I think we can just find-replace the `// MOCK FILE!` comment with a more detailed description.